### PR TITLE
get all products data

### LINKED ISSUE #8: Back-end Route| GET |  /products
#### Products  Route
 get all products information in json format from the database
#### Protected: false

---

#### Main Things : 
- [x] Database
- [x] Response
- [x] Test

---

#### Database :
Use database query to fetch all the data  related to products 

---
#### Response : 
Response may be success or error :
* Error :
```javascript
    {
        status:400,
        messgae:'Bad reqeust'
    }
```
* Success
```javascript
    {
        status:200,
        data: [
         {id:1, title: 'قميص ماركة بولو',.....},
         {id:1, title: 'حذاء ماركة اديدس',.....}
         ] , 
    }
```
---
#### Test : 
* Route : 
  Test route by testing all the expected cases for correct retrieve for data from application table.
  
* Query : 
  Test select database query to make sure that everything is ok.

### PR DESCRIPTION
**Pull request by AlaaShurrab**
fixes #8 
creating the route to get all the products from the database

### CODE REVIEW

File: server/database/quieres/getAllProducts.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const connection = require('../connection');
+
+const getAllProducts = () => connection.query('select id, name, img_url, rating, old_price, new_price from products ;');
```
Comment (mFlaifel) [inline server/database/quieres/getAllProducts.js]: please add a product description, quantity, and category
and don't forget to modify your test.

**Review by mFlaifel — CHANGES_REQUESTED** [review]
good job

File: server/database/quieres/getAllProducts.js (hunk 1 above)
Comment (AlaaShurrab) [reply server/database/quieres/getAllProducts.js]: done

File: __test__/getAllProducts.test.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+const app = require('../server/app');
+const connection = require('../server/database/connection');
+const buildDb = require('../server/database/data/build');
+const { getAllProducts } = require('../server/database/quieres');
+
+describe('Test Routes, Test Database getClients query', () => {
+  beforeEach(() => buildDb());
+  afterAll(() => connection.end());
+
+  test('getAllProducts >> This test will return clients data', () => getAllProducts()
```
Comment (Sondos-Alalami) [inline __test__/getAllProducts.test.js]: please modify describe and test texts, remove clients and getClients.

**Review by Sondos-Alalami — CHANGES_REQUESTED** [review]

File: __test__/getAllProducts.test.js (hunk 2 above)
Comment (AlaaShurrab) [reply __test__/getAllProducts.test.js]: done

File: server/controllers/common/handlerAllProducts.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const { getAllProducts } = require('../../database/quieres');
```
Comment (mFlaifel) [inline server/controllers/common/handlerAllProducts.js]: there is a typo in quieres ,
change it to 'queries' and modify all related files and folder

**Review by mFlaifel — CHANGES_REQUESTED** [review]

**Review by mFlaifel — APPROVED** [review]

**Pull request merged**